### PR TITLE
add filters according to date range, Created Hotel model and APIs, As…

### PR DESCRIPTION
**Pull request by harmeshverma01**
… a particular hotel is going to have rooms

### CODE REVIEW

File: room/models.py [hunk 1]
```diff
@@ -12,7 +10,18 @@ class Room(models.Model):
     image = models.ImageField()
     floor = models.IntegerField()
     user = models.ForeignKey(to=User, on_delete=models.CASCADE, related_name='User', null=True)
+    bad_required = models.IntegerField(null=True)
```
Comment (TACGithub) [inline room/models.py:13]: It should be bed_count.

File: room/models.py [hunk 2]
```diff
@@ -12,7 +10,18 @@ class Room(models.Model):
...
     
     def __str__(self) -> str:
         return self.name
-    
\ No newline at end of file
+    
+
+class Hotel(models.Model):
+    name = models.CharField(max_length=150)
+    room = models.ForeignKey(to=Room, on_delete=models.CASCADE, related_name='Room', null=True)
```
Comment (TACGithub) [inline room/models.py:23]: Explain why hotel has the foreign key of room instead of Room having the hotel foreign key.

File: room/views.py [hunk 3]
```diff
@@ -1,37 +1,43 @@
...
 
 class RoomView(APIView):
     serializer_class = RoomSerializer
+    permission_classes = [admin_required]
     
     def get(self, request, id=None):
         user = Room.objects.all()
+        price = request.GET.get('price', None)
+        if price is not None:
+            user = user.filter(price=price)
```
Comment (TACGithub) [inline room/views.py:18]: price should be a range, not an actual number

File: room/views.py [hunk 4]
```diff
@@ -1,37 +1,43 @@
...
+                    
 class RoombookedView(APIView):
     serializer_class = RoomSerializer
+    permission_classes = [admin_required]
     
     def get(self, request, id=None):
         user = Room.objects.all()
+        date = request.GET.get('date', None)
+        if date is not None:
+            user = user.filter(date=date)
```
Comment (TACGithub) [inline room/views.py:31]: Date filter should be a range also.

File: room/models.py (hunk 1 above)
Comment (harmeshverma01) [reply room/models.py:13]: fixed

File: room/models.py (hunk 2 above)
Comment (harmeshverma01) [reply room/models.py:23]: Its is not in room model it is in Hotel model which having a foreignkey instead. Fixed

File: room/views.py (hunk 3 above)
Comment (harmeshverma01) [reply room/views.py:18]: fixed

File: room/views.py (hunk 4 above)
Comment (harmeshverma01) [reply room/views.py:31]: fixed

**Pull request merged**